deepmd/pt_expt/train/training.py

```
@@ -557,58 +729,125 @@ def get_sample() -> list[dict[str, np.ndarray]]:
+                            self._broadcast_model_stat(self.model[model_key])
             else:
                 # --- Normal resume (init_model / restart) --------------------
-                self.wrapper.load_state_dict(state_dict)
+                self._unwrapped.load_state_dict(state_dict)
+
+            if shared_links is not None:
+                # Re-apply sharing after loading checkpoint
+                self._unwrapped.share_params(
+                    shared_links,
+                    resume=True,
+                    model_key_prob_map=dict(zip(self.model_keys, self.model_prob)),
```

Ruff (0.15.10)

[warning] `zip()` without an explicit `strict=` parameter

Add explicit value for parameter `strict=`

(B905)